docs/API.md

```diff
@@ -106,3 +108,67 @@ resolvers = {
 }
 ```
 4. **directive_resolvers:** Resolver list which is used as the directive resolvers.
+
+## Resolver
+
+### Declare resolvers
+
+Tartiflette provides 2 different ways to declare a resolver for a Field.
+
+#### With decorators
+
+The most common way to asign a specific resolver for a Field is to put a decorator on top of the resolver function, that one [MUST BE compliant with the function signature](#function-signature).
+
+```python
+from tartiflette import Resolver
+
+@Resolver("Query.hello")
+def my_hello_resolver(parent, args, context, info):
+    return "Chuck"
+```
+
+#### During Engine initialization
+
+The second way is to use the [advanced Engine initialization](#advanced-constructor) to specify a list of resolvers associated to Fields.
+
+```python
+import tartiflette
+
+def my_hello_resolver(parent, args, ctx, info):
+    return "Chuck"
+
+engine = tartiflette.Engine(
+    sdl,
+    resolvers={
+        "Query.hello": my_hello_resolver
+    }
+)
+```
+
+### Function signature
+
+Every resolver in Tartiflette accepts four positional arguments:
+
+_(This signature has been highly inspired by the GraphQL.js implementation)_
+
+```python
+def my_hello_resolver(parent, args, context, info):
+    pass
+```
+
+1. **parent:** Object which contains the result returned by the resolver of the parent field. The `root_value` is passed in the case of a top-level Query field.
+2. **args:** A dict which contains the arguments passed for the field. _(in the query)_. e.g. if the field was called with `hello(name: "Chuck")`, the args dict will be equals to `{ "name": "Chuck" }`.
+3. **context:** Dict shared by all resolvers, could be different for each query. It acts-as a container or a state for a specific request.
```

Comment:
acts as (no dash)
or a state <= what do you mean ? Isn't it as a store/storage ?

Reply:
`acts as (no dash)`

Fixed

`or a state <= what do you mean ? Isn't it as a store/storage ?`

We will let the developers chose what they want to do. They could use the context as a container (like us at dailymotion) or like a state ...